linker.go

```diff
@@ -61,7 +61,11 @@ func needSection(insns, section asm.Instructions) (bool, error) {
 			continue
 		}
 
-		if ins.OpCode.JumpOp() != asm.Call || ins.Src != asm.PseudoCall {
```

Comment:
This can be rewritten as `!ins.IsFunctionCall()`. Then you can use `!ins.IsFunctionCall() && !ins.IsLoadOfFunctionPointer()` instead of the two ifs.